src/main/java/seedu/apollo/command/module/AddModuleCommand.java

```diff
@@ -228,6 +236,67 @@ public ArrayList<LessonType> getLessonTypes(Module module) {
         return lessonTypes;
     }
 
+    /**
+     * Checks if the lesson clashes with another lesson.
+     *
+     * @param calendar The calendar of the user containing timetable information.
+     * @param timetable The timetable of the lesson to be checked.
+     * @param ui The ui of the user for message printing.
+     */
+    private void checkClashingLesson(Calendar calendar, Timetable timetable, Ui ui) {
+        String day = timetable.getDay();
+        int index = determineDay(day);
+
+        if (index == -1) {
+            return;
+        }
+
+        if (calendar.get(index).size() == 0) {
+            return;
+        }
+
+        for (CalendarModule lessonModule: calendar.get(index)) {
+            Timetable schedule = lessonModule.getSchedule();
+            if (isClashing(schedule, timetable)) {
+                ui.printClashingLesson();
+                break;
+            }
+        }
```

Comment:
Maybe make this throw a LessonClashException that gets tackled in a higher layer so ui doesn't need to be passed through all the way

Reply:
Wah can but I was lazy